### PR TITLE
Autofixing offenses from current config

### PR DESCRIPTION
**Pull request by cinkonaap**
Provided functionality to autofix specific offenses against provided directories (using AST provided by `gonzales-pe`).

For now I've implemented rules that happens often in our target app, which are:

* `border-zero`
* `empty-line-between-blocks`
* `hex-length`
* `hex-notation`
* `leading-zero`
* `no-trailing-zero`
* `quotes`
* `shorthand-values`
* `space-after-colon`
* `space-after-comma`
* `space-before-brace`
* `space-before-colon`
* `zero-unit`

You can run it from your app by running `js script` with parameters:

`node node_modules/sass-lint-config-airhelp/autofix/index.js --dir dir1 dir2 --rules hex-notation hex-length`

### CODE REVIEW

File: autofix/index.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+const gonzales = require('gonzales-pe');
+const fs = require('fs');
+const path = require('path');
+const walkSync = require('walk-sync');
+const yargs = require('yargs');
+const rulesMethods = require('./rules');
+
+const { log } = console;
+const argv = yargs.array('dir').array('rules').argv;
+let { rules, dir } = argv;
```
Comment (jedrula) [inline autofix/index.js]: dir could probably be a const

File: autofix/index.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+rules.forEach(rule => {
+  fixStatistics[rule] = 0;
+});
+
+// process fixing for each file
+scssFiles.forEach(file => {
+  const fullPath = path.join(root, file);
+  const contents = fs.readFileSync(fullPath, { encoding: 'utf8' });
+
+  if (contents.length == 0) {
```
Comment (jedrula) [inline autofix/index.js]: why do you have to guard against empty contents?

File: autofix/index.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  const parsedTree = gonzales.parse(contents, { syntax: 'scss' });
+
+  // for each rule, call with current file contents
+  rules.forEach(rule => {
+    const fixedCount = rulesMethods[rule](parsedTree);
+    fixStatistics[rule] += fixedCount;
+  });
+
+  // write fixed content to file
+  if (argv.ns !== true) {
```
Comment (jedrula) [inline autofix/index.js]: what is `argv.ns` ?

Comment (cinkonaap) [inline autofix/index.js]: It's `no-save` options. I haven't found way to use full `--no-save` param, but I will be able to comment this properly with `yargs`

File: autofix/index.js (hunk 2 above)
Comment (cinkonaap) [inline autofix/index.js]: First of all we haven't to run any additional code in such cases, but secondly it was protecting from some strange things (empty file was converted to `undefined`)

**Comment by jedrula** [comment]
👍

**Comment by jedrula** [comment]
well done!

**Pull request merged**